guacamole/src/main/frontend/src/app/import/types/DisplayErrorList.js

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * Service which defines the DisplayErrorList class.
+ */
+angular.module('import').factory('DisplayErrorList', [
+        function defineDisplayErrorList() {
+
+    /**
+     * A list of human-readable error messages, intended to be usable in a
+     * sortable / filterable table.
+     *
+     * @constructor
+     * @param {String[]} messages
+     *     The error messages that should be prepared for display.
+     */
+    const DisplayErrorList = function DisplayErrorList(messages) {
+
+        // Use empty message list by default
+        this.messages = messages || [];
+
+        // The single String message composed of all messages concatenated
+        // together. This will be used for filtering / sorting, and should only
+        // be calculated once.
+        this.cachedMessage = null;
```

Comment:
This doesn't appear to be used, though there is a reference to `concatenatedMessage`.

Reply:
Oops, that's supposed to be `concatenatedMessage`.